Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -187,6 +187,9 @@ void ViewTransition::skipViewTransition(ExceptionOr<JSC::JSValue>&& reason)
     }
 
     m_updateCallbackDone.first->whenSettled([this, protectedThis = Ref { *this }] {
+        if (m_updateCallbackDone.first->isSuspended())
```

Comment:
Checking if the ViewTransition object `isStopped()` is the more common pattern.

Reply:
Does it work to push this check into `whenPromiseIsSettled`? I assume it would be annoying/error-prone to remember that documents can get detached whenever you handle a promise in WebCore.

Reply:
> Checking if the ViewTransition object `isStopped()` is the more common pattern.

ViewTransition doesn't have `isStopped()` method. Do you mean `ActiveDOMObject::isContextStopped()`?

Reply:
> Does it work to push this check into `whenPromiseIsSettled`? I assume it would be annoying/error-prone to remember that documents can get detached whenever you handle a promise in WebCore.

What are you suggesting? Don't call the callback at all if the promise is suspended? Could there be cases where we do something in the callback regardless of the status of the promise?